pygfx/helpers/_stats.py

```diff
@@ -0,0 +1,141 @@
+    ----------
+    viewport : Viewport, Renderer
+        Required for positioning lines of text in the corner of
+        the screen.
+
+    """
+
+    def __init__(
+        self,
+        viewport,
+    ):
+        from .. import Viewport  # avoid circular import
+
+        super().__init__()
+
+        self._line_height = 16
+        font_size = 12
+        foreground = "#0f0"
+        background = "#020"
+        text_material = TextMaterial(color=foreground)
+
+        quad_geometry = plane_geometry()
+        quad_geometry.positions.data[..., :2] += [0.5, -0.5]
+
+        self.bg = Mesh(
+            quad_geometry,
+            MeshBasicMaterial(
+                color=background,
+                side="BOTH",
+                opacity=0.9,
+            ),
+        )
+        # refactor once text bounding boxes are available
+        self.bg.scale.set(90, self._line_height * 2.1, 1)
+        self.ms = Text(
+            TextGeometry(
+                text="",
+                screen_space=True,
+                font_size=font_size,
+                anchor="topleft",
+            ),
+            text_material,
+        )
+        self.fps = Text(
+            TextGeometry(
+                text="",
+                screen_space=True,
+                font_size=font_size,
+                anchor="topleft",
+            ),
+            text_material,
+        )
+        self.add(self.bg, self.ms, self.fps)
+
+        self.camera = ScreenCoordsCamera()
+
+        # track screen size for line positioning
+        self._viewport = Viewport.from_viewport_or_renderer(viewport)
+        self._update_positions()
+        self._viewport.renderer.add_event_handler(
+            self._update_positions,
+            "resize",
+        )
+
+        # flag used to skip the first frame
+        # which typically has all the CPU->GPU transfer and
+        # shader compilation overhead
+        self._init = False
+
+        # performance trackers
+        self._tmin = 1e10
+        self._tmax = 0
+        self._tbegin = None
+        self._tprev = self._tbegin
+        self._frames = 0
+        self._fmin = 1e10
+        self._fmax = 0
+
+    def _update_positions(self, event=None):
+        _, height = self._viewport.logical_size
+        self.ms.position.set(0, height, 0)
+        self.fps.position.set(0, height - self._line_height, 0)
+        self.bg.position.set(0, height, 0.1)
+
+    def start(self):
+        if not self._init:
+            return
+
+        self._tbegin = time.perf_counter_ns()
+        if self._tprev is None:
+            self._tprev = self._tbegin
+
+    def stop(self):
+        if not self._init:
+            self._init = True
+            return
+
+        t = time.perf_counter_ns()
+        self._frames += 1
+
+        delta = round((t - self._tbegin) / 1_000_000)
+        self._tmin = min(self._tmin, delta)
+        self._tmax = max(self._tmax, delta)
+        self.ms.geometry.set_text(f"{delta} MS ({self._tmin}-{self._tmax})")
```

Comment:
This now reads as Mega-seconds 😉 , let's use "ms". Also, since the numbers would often be in the order of a few ms, would it it make sense to show one decimal?

Reply:
Ok 👍 let's try like this.